ts/components/conversation/media-gallery/MediaGallery.tsx

```diff
@@ -0,0 +1,148 @@
+/**
+ * @prettier
+ */
+import React from 'react';
+
+import moment from 'moment';
+
+import { AttachmentSection } from './AttachmentSection';
+import { groupMessagesByDate } from './groupMessagesByDate';
+import { ItemClickEvent } from './events/ItemClickEvent';
+import { Message } from './propTypes/Message';
+
+type AttachmentType = 'media' | 'documents';
+
+interface Props {
+  documents: Array<Message>;
+  i18n: (key: string, values?: Array<string>) => string;
+  media: Array<Message>;
+  onItemClick?: (event: ItemClickEvent) => void;
+}
+
+interface State {
+  selectedTab: AttachmentType;
+}
+
+const MONTH_FORMAT = 'MMMM YYYY';
+const COLOR_GREY = '#f3f3f3';
+
+const tabStyle = {
+  width: '100%',
+  backgroundColor: COLOR_GREY,
+  padding: 20,
+  textAlign: 'center',
+};
+
+const styles = {
+  tabContainer: {
+    cursor: 'pointer',
+    display: 'flex',
+    width: '100%',
+  },
+  tab: {
+    default: tabStyle,
+    active: {
+      ...tabStyle,
+      borderBottom: '2px solid #08f',
+    },
+  },
+  attachmentsContainer: {
+    padding: 20,
+  },
+};
+
+interface TabSelectEvent {
+  type: AttachmentType;
+}
+
+const Tab = ({
```

Comment:
For a small component like this we have three options:

1. make it a separate component in the same file
2. make a separate component in a different file
3. make it a `renderXXX` method of the component using it

How do you choose between those three?

Reply:
I prefer (1) which can then easily be moved to (2) if it gets too big.